### PR TITLE
Removes transition btc->bat code

### LINKED ISSUE #12712: phase two transitioning from the BTC proof-of-concept to BAT
when `init` is called in `ledger.js` if the `statePath` file exists and is for the BTC proof-of-concept, immediately rename it to another name.

thus, when someone starts the ledger up after a long period of inactivity (it's been over three months), they'll get a fresh initialization vector.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #12712

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

1) old wallet
- create wallet on version 0.18 so that you have BTC wallet
- copy that wallet into this branch
- run browser
- check payments page (you should see new wallet created for you)
- check profile folder to see if there is file `ledger-state-btc.json`

2) new wallet
- create wallet on 0.20
- add some BAT into it
- copy that wallet into this branch
- run browser
- make sure that no money was lost and that old wallet is still there

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
@NejcZdovc can we get a rebase on this one?

**Comment by NejcZdovc** [comment]
@bsclifton rebased

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -2390,147 +2388,6 @@ const deleteSynopsis = () => {
   synopsis.publishers = {}
 }
 
-// fix for incorrectly persisted state (see #11585)
-const yoDawg = (stateState) => {
```
Comment (bsclifton) [inline app/browser/api/ledger.js]: RIP the yoDawg bug fix 💯 
cc: @evq

File: app/browser/api/ledger.js [hunk 2]
```diff
@@ -2072,6 +2054,22 @@ const getContributionAmount = (state) => {
 }
 
 const onInitRead = (state, parsedData) => {
+  const isBTC = parsedData &&
+    parsedData.properties &&
+    parsedData.properties.wallet &&
+    parsedData.properties.wallet.keychains
+
+  if (isBTC) {
```
Comment (bsclifton) [inline app/browser/api/ledger.js]: I think we should add verbose logging here- something like:
`Bitcoin wallet detected; backing up to "ledger-state-btc.json" and creating new BAT wallet`

**Comment by bsclifton** [comment]
- I looked over the code changes closely and they look great 👍 
- Test plan 1 worked great 👍 
- Test plan 2 is failing for me... I copied my production profile (both `session-store-1` and `ledger-state.json`) and the following error is logged to console:
```
An uncaught exception occurred in the main process Uncaught Exception:
TypeError: First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.
    at fromObject (buffer.js:274:9)
    at Function.Buffer.from (buffer.js:106:10)
    at Object.Client.getWalletPassphrase (/Users/clifton/Documents/browser-laptop/node_modules/bat-client/index.js:508:46)
    at getStateInfo (/Users/clifton/Documents/browser-laptop/app/browser/api/ledger.js:1524:43)
    at Object.onInitRead (/Users/clifton/Documents/browser-laptop/app/browser/api/ledger.js:2079:11)
    at ledgerReducer (/Users/clifton/Documents/browser-laptop/app/browser/reducers/ledgerReducer.js:302:27)
    at handleAppAction (/Users/clifton/Documents/browser-laptop/js/stores/appStore.js:230:16)
    at callbacks.forEach (/Users/clifton/Documents/browser-laptop/js/dispatcher/appDispatcher.js:107:7)
    at Array.forEach (<anonymous>)
    at AppDispatcher.dispatchToOwnRegisteredCallbacks (/Users/clifton/Documents/browser-laptop/js/dispatcher/appDispatcher.js:106:20)
    at AppDispatcher.dispatchInternal (/Users/clifton/Documents/browser-laptop/js/dispatcher/appDispatcher.js:132:10)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```

**Comment by NejcZdovc** [comment]
@bsclifton second test plan error is not related to this PR. Will be fixed with this one https://github.com/brave/browser-laptop/pull/13201

**Comment by bsclifton** [comment]
@evq @LaurenWags @srirambv would you all care to give this one a review too? 😄

**Comment by LaurenWags** [comment]
Tests Ran:

1. Old Profile and (production) wallet from 0.18.36 (last released version with BTC wallets) Payments were ON and my ledger table had data - PASSED
a. took a copy of profile from 0.18.36 where Payments were ON and ledger table had data (wallet had 0 BTC), renamed the profile copy to be brave-development
b. built and ran PR
c. Verified did not see transition overlay (bouncing balls), verified ledger data was retained, verified wallet was created (addresses and QR codes visible in 'Add Funds'), verified presence of `ledger-state-btc.json` file in brave-development profile folder.

2. Old Profile and (production) wallet from 0.18.36 (last released version with BTC wallets) Payments were OFF and my ledger table had data - PASSED
a. took a copy of profile from 0.18.36 where Payments were OFF and ledger table had data (wallet had 0 BTC), renamed the profile copy to be brave-development
b. built and ran PR
c. Enabled Payments from Preferences
d. Verified did not see transition overlay (bouncing balls), verified ledger data was retained, verified wallet was created (addresses and QR codes visible in 'Add Funds'), verified presence of `ledger-state-btc.json` file in brave-development profile folder.

3. Profile and (staging) wallet from 0.20.42. Payments were ON and my ledger table had data
a. took a copy of a profile from 0.20.42 where Payments were ON and ledger table had data (staging wallet had X BAT), renamed the profile copy to be brave-development

**Comment by bsclifton** [comment]
master aedf064f922261e564451ec4e01ec26849e5c196

@NejcZdovc will need your help cherry-picking to 0.22.x and 0.21.x 😄

**Comment by NejcZdovc** [comment]
0.22 b68673f9c
0.21 ff3e0ec98

**Comment by NejcZdovc** [comment]
0.21 twitch 9f85d9f8d

**Pull request merged**